### PR TITLE
ID-241 ID-244 override with env conf

### PR DESCRIPTION
**Pull request by tlangs**
Tickets: https://broadworkbench.atlassian.net/browse/ID-241 https://broadworkbench.atlassian.net/browse/ID-244

Why: firecloud-develop bad. Configuring with terra-helmfile environment variables good.

How:  
HOCON supports combining conf files with fallbacks, which means if a value is not defined in `resource-a.conf`,but is defined in `resource-b.conf`, then `ConfigFactory.load("resource-a").withFallback(ConfigFactory.load("resource-b"))`
will contain values from both resource files. For any variable in both, that variable will be populated by `resource-a.conf`. Additionally, we can provide environment variables to populate the resource file by defining `db.sam_read.user=${?SAM_READ_USER}`, which will only load the value if the environment variable is defined. With this, we can define an `env.conf`, filled with optional environment variables, and load that with the default config as a fallback, meaning environment variables that are defined will be read, and those that aren't will be populated by the default conf file.

We've also made `AppConfig` handle this conf file reading internally so that it can be tested.

---

**PR checklist**

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've filled out the [Security Risk Assessment](https://broadinstitute.github.io/dsp-appsec-security-risk-assessment/) and attached the result to the JIRA ticket

### CODE REVIEW

File: ENV_FALLBACK.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Configuring SAM with Environment Variables
+
+HOCON supports combining conf files with fallbacks, which means if a value is not defined in `resource-a.conf`,
+but is defined in `resource-b.conf`, then `ConfigFactory.load("resource-a").withFallback(ConfigFactory.load("resource-b"))`
+will contain values from both resource files. For any variable in both, that variable will be populated by `resource-a.conf`.
+Additionally, we can provide environment variables to populate the resource
+file by defining `db.sam_read.user=${?SAM_READ_USER}`, which will only load the value if the environment variable is defined.
+With this, we can define an `env.conf`, filled with optional environment variables, and load that with the default config
+as a fallback, meaning environment variables that are defined will be read, and those that aren't will be populated by
+the default conf file.
```
Comment (Ghost-in-a-Jar) [inline ENV_FALLBACK.md]: lets link this file in readme or move the text there

**Review by andy7i — APPROVED** [review]
👍

**Review by Ghost-in-a-Jar — APPROVED** [review]
Mobbed! 🥇

**Comment by tlangs** [comment]
Jenkins retest

**Comment by tlangs** [comment]
jenkins retest

**Comment by tlangs** [comment]
jenkins retest

**Comment by tlangs** [comment]
jenkins retest

**Pull request merged**